core/src/main/java/org/wildfly/extras/sunstone/api/impl/azure/AzureNode.java

```diff
@@ -102,7 +102,13 @@ public AzureNode(AzureCloudProvider azureCloudProvider, String name, Map<String,
 
         int timeout = objectProperties.getPropertyAsInt(nodeConfigData.waitForPortsTimeoutProperty,
                 nodeConfigData.waitForPortsDefaultTimeout);
-        waitForPorts(timeout, 22);
+        try {
+            waitForPorts(timeout, 22);
+        } catch (Exception e) {
+            this.close();
```

Comment:
IMO, this is not correct. As the object is not yet constructed and we provide a reference to it to another object (`CloudProvider.destroyNode(this)`). We should work directly with JClouds API here. Sth like

```java
computeService.destroyNode(initialNodeMetadata.getId());
```
And probably to take in account also the `leaveNodesRunning` property from the cloud provider.